chips/lowrisc/src/aon_timer.rs

```diff
@@ -0,0 +1,232 @@
+            registers: base,
+            aon_clk_freq: aon_clk_freq,
+        }
+    }
+
+    /// Reset both watch dog and wake up timer count values.
+    fn reset_timers(&self) {
+        let regs = self.registers;
+        regs.wkup_count.set(0x00);
+        regs.wdog_count.set(0x00);
+    }
+
+    /// Start the watchdog counter with pause in sleep
+    /// i.e wdog timer is paused when system is sleeping
+    fn wdog_start_count(&self) {
+        self.registers
+            .wdog_ctrl
+            .write(WDOG_CTRL::ENABLE::SET + WDOG_CTRL::PAUSE_IN_SLEEP::SET);
+    }
+
+    /// Program the desired thresholds in WKUP_THOLD, WDOG_BARK_THOLD and WDOG_BITE_THOLD
+    fn set_wdog_thresh(&self) {
+        let regs = self.registers;
+        // Watchdog period may need to be revised with kernel changes/updates
+        // since the watchdog is `tickled()` at the start of every kernel loop
+        // see: https://github.com/tock/tock/blob/eb3f7ce59434b7ac1b77ef1ab7dd2afad1a62ac5/kernel/src/kernel.rs#L448
+        let bark_cycles = self.ms_to_cycles(500);
+        // ~1000ms bite period
+        let bite_cycles = bark_cycles.saturating_mul(2);
+
+        regs.wdog_bark_thold
+            .write(THRESHOLD::THRESHOLD.val(bark_cycles));
+        regs.wdog_bite_thold
+            .write(THRESHOLD::THRESHOLD.val(bite_cycles));
+    }
+
+    // Reset watch dog timer
+    fn wdog_pet(&self) {
+        self.registers.wdog_count.set(0x00);
+    }
+
+    fn wdog_suspend(&self) {
+        self.registers.wdog_ctrl.write(WDOG_CTRL::ENABLE::CLEAR);
+    }
+
+    fn wdog_resume(&self) {
+        self.registers.wdog_ctrl.write(WDOG_CTRL::ENABLE::SET);
+    }
+
+    /// Locks further config to WDOG until next system reset
+    fn lock_wdog_conf(&self) {
+        self.registers.wdog_regwen.write(WDOG_REGWEN::REGWEN::SET)
+    }
+
+    /// Convert microseconds to cycles
+    fn ms_to_cycles(&self, ms: u32) -> u32 {
+        // 250kHZ CW130 or 125kHz Verilator (as specified in chip config)
+        ms.saturating_mul(self.aon_clk_freq).saturating_div(1000)
+    }
+
+    /// Start the wake up counter
+    #[cfg(aon_wkup_timer)]
+    fn wkup_start_count(&self) {
+        self.registers.wkup_ctrl.write(WKUP_CTRL::ENABLE::SET);
+    }
+
+    /// Set wake-up prescaler, NOTE: Independant of the watchdog timer
+    /// A count starts at 0 and slowly ticks upwards
+    /// (one tick every N + 1 clock cycles, where N is the pre-scaler value)
+    #[cfg(aon_wkup_timer)]
+    fn set_wkup_prescaler(&self) {
+        // Ex: AON VerilatorClock ~125kHZ, (2^12)/125_000Hz ~= 32ms (per tick)
+        // NOTE: Arbitrarily chosen, update as necessary.
+        let prescaler = 0x1000; // 2^12
+        self.registers
+            .wkup_ctrl
+            .write(WKUP_CTRL::PRESCALER.val(prescaler));
+    }
+
+    /// Set threshold ticks for the wake-up interrupt,
+    /// Note: the timing here depends on the prescaler set in
+    /// `set_wkup_prescaler()`
+    #[cfg(aon_wkup_timer)]
```

Comment:
What is this `cfg` for?

Reply:
Since we aren't using the `aon wakeup timer` it's just there to compile out this code. I left the code there for reference, perhaps if we end up implementing this timer later, you could remove the tags and call it.

Reply:
Ha, I should have read the description